service/src/main.rs

```diff
@@ -334,6 +336,42 @@ fn start_worker<E, T, D>(
 	let mut xthex = hex::encode(uxt);
 	xthex.insert_str(0, "0x");
 
+	//Account funds
```

Comment:
Maybe we should put this entire part into a separate function - `account_funding()` or similar. Otherwise it really clutters this already too long main function. And you could save the comment, because the name of the function would be enough.